### PR TITLE
Animations

### PR DESCRIPTION
**Pull request by NoiSek**
Does a lot of things, doesn't do some other things. This PR introduces the barebones elements of an animation system, adds further debug tools, fixes quite a few bugs, carries a laundry list of minor changes, and introduces a more efficient (and flexible) pathing solution.

As this branch has been growing increasingly stale and unfocused with it's intent as time goes on, I'm breaking early and merging this back into master. Will also be deferring further Dungeons work into the Unity prototype for efficiency reasons until further notice.

### Migration Notes

- Fixes a locked foreign-key issue in pets when using the provisioner, DB may need to be destroyed and rebuilt in order to reconcile the changes.
- Ubuntu image has been changed to a more stable build, Vagrant users will need to destroy and rebuild their local environments (and boxes)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,6 +9,7 @@ node_modules
 .vagrant
 build
 npm-debug.log
+package-lock.json
```
Comment (tiffz) [inline .gitignore]: Wait, isn't it explicitly recommended to version control the package-lock file haha? (I do always just constantly delete it and recreate it though ehhee.)

File: gulpfile.babel.js [hunk 2]
```diff
@@ -10,21 +10,22 @@ import autoprefixer from 'gulp-autoprefixer';
...
 import webpack from 'webpack';
+import cloneDeep from 'lodash.clonedeep';
 
 import webpackConfig from './webpack.config.js';
 import tempConfig from './temporary.config.js';
 
 const paths = tempConfig.paths;
 const dests = tempConfig.dests;
 
+
```
Comment (tiffz) [inline gulpfile.babel.js]: I'm going to add a comment on this whitespace because I can.

**Review by LikeMyBread — APPROVED** [review]
Looks good to me, thanks for the quick fix on the failing tests. I'm thinking we probably won't push this to production right away as there's nothing that looks to need immediate beta testing.

File: saylua/modules/adventure/static-source/es6/Dungeon/Core/GameRenderer.js [hunk 3]
```diff
@@ -244,7 +231,7 @@ export default class GameRenderer {
 
 
   test() {
-
+    console.log("Hello! I am the test!");
```
Comment (tiffz) [inline saylua/modules/adventure/static-source/es6/Dungeon/Core/GameRenderer.js]: Hello, I am Tiff!!

**Review by tiffz — APPROVED** [review]
Awesome, lgtm!

File: .gitignore (hunk 1 above)
Comment (NoiSek) [reply .gitignore]: I hadn't realized newer versions of npm generated this, actually, I always thought it was just a byproduct of yarn (which I was under the impression neither of you used)

**Pull request merged**